### ISSUE TITLE
cc65: Invalid parameter handling for K&R functions

### ISSUE THREAD

**Issue by kugelfuhr** [cc65/cc65#2300]
The following code should compile but doesn't:

```C
int foo(a, b) { return a + b; }

int main()
{
    foo(1, 2, 3);
    return 0;
}
```

Gives:
```
test.c:5: Error: Too many arguments in function call
```

This is related to the problem described in #2014: The compiler checks the parameter list when a K&R function is called, but it shouldn't. The information given by the function definition 
```
int foo(a, b) { return a + b; }
```
is equivalent to
```
int foo();
```

**Comment by acqn** [comment]
> If the expression that denotes the called function has _a type that does not include a prototype_, the
integer promotions are performed on each argument, and arguments that have type float are
promoted to double. These are called the default argument promotions. _If the number of arguments
does not equal the number of parameters, **the behavior is undefined**._

So I think the compiler can just reject the code according to the standard.

**Comment by acqn** [comment]
OK. So it is actually easier than I thought to implement the "K&R function definition as empty-parameter-list function declaration", and it makes some other things easier.

Still there are the questions:
When there is indeed a K&R function definition, should we give diagnosis on a call to the function if something is wrong with the arguments eg. incompatible type, wrong count etc.? If we would like to accept such code, should we at least give a warning on it? And if the pushed size of the argument could be wrong, should we try to "fix" the stack, or just compile and let it blow up?

**Comment by GorillaSapiens** [comment]
i don't think this should really be a bug.  where ever is such broken behavior useful?

**Comment by kugelfuhr** [comment]
> i don't think this should really be a bug. where ever is such broken behavior useful?

Acceptance of such code is a requirement according to ISO/IEC 9899:1990(E) and ISO/IEC 9899:1999(E).
